src/svg.c

```diff
@@ -80,13 +88,14 @@ int svg_to_ps(lua_State *L) {
           output[output_l++] = ' ';
         }
       }
-      if (output_l + 3 > max_output) {
-        output = realloc(output, max_output + 3);
-      }
-      output[output_l++] = strokeFillOper;
-      output[output_l++] = ' ';
-      output[output_l] = '\0';
     }
+    for (char* c = strokeFillOper; *c != '\0'; c++)
+        output[output_l++] = *c;
+    if (output_l + 4 > max_output) {
```

Comment:
It's fine to just increase the margin by 1 here but if there's insufficient space you need to realloc before copying strokeFillOper into the output, not after.

Rather than adding custom code to copy the string you might just call `safe_append()` instead?